proc.c

```diff
@@ -1437,8 +1438,24 @@ rb_hash_proc(st_index_t hash, VALUE prc)
 {
     rb_proc_t *proc;
     GetProcPtr(prc, proc);
-    hash = rb_hash_uint(hash, (st_index_t)proc->block.as.captured.code.val);
-    hash = rb_hash_uint(hash, (st_index_t)proc->block.as.captured.self);
+
+    switch (vm_block_type(&proc->block)) {
```

Comment:
Any reason to have a `rb_hash_proc ` instead of just falling back to `rb_obj_hash`? I guess it saves generating an object id?

Reply:
TIL, carry on then :)